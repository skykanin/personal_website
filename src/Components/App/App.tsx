import * as React from 'react';

import styled from 'styled-components';
import logo from '../../logo.svg';

const Header = styled.div`
  background-color: papayawhip;
  height: 150px;
  padding: 20px;
  color: palevioletred;
`;

const Root = styled.div`
  text-align: center;
`;

const Logo = styled.img`
  animation: App-logo-spin infinite 20s linear;
  height: 80px;

  @keyframes App-logo-spin {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
  }

`;

const Title = styled.h1`
  font-size: 1.5em;
`;

const Intro = styled.p`
  font-size: large;
`;

export default class App extends React.Component {
  public render() {
    return (
      <Root>
        <Header>
          <Logo src={logo} alt="logo" />
          <Title>Welcome to React</Title>
        </Header>
        <Intro>
          To get started, edit <code>src/App.tsx</code> and save to reload.
        </Intro>
      </Root>
    );
  }
}
