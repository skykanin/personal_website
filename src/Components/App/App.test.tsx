import { shallow } from 'enzyme';
import * as React from 'react';

import App from './App';

it('renders all elements correctly', () => {
  const appComponent = shallow(<App />)
  expect(appComponent).toMatchSnapshot();
});
