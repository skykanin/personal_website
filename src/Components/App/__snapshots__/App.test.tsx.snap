// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all elements correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.img
            alt="logo"
            src={undefined}
          />
          <styled.h1>
            Welcome to React
          </styled.h1>
        </styled.div>,
        <styled.p>
          To get started, edit 
          <code>
            src/App.tsx
          </code>
           and save to reload.
        </styled.p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.img
              alt="logo"
              src={undefined}
            />,
            <styled.h1>
              Welcome to React
            </styled.h1>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alt": "logo",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Welcome to React",
            },
            "ref": null,
            "rendered": "Welcome to React",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "To get started, edit ",
            <code>
              src/App.tsx
            </code>,
            " and save to reload.",
          ],
        },
        "ref": null,
        "rendered": Array [
          "To get started, edit ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "src/App.tsx",
            },
            "ref": null,
            "rendered": "src/App.tsx",
            "type": "code",
          },
          " and save to reload.",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.img
              alt="logo"
              src={undefined}
            />
            <styled.h1>
              Welcome to React
            </styled.h1>
          </styled.div>,
          <styled.p>
            To get started, edit 
            <code>
              src/App.tsx
            </code>
             and save to reload.
          </styled.p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.img
                alt="logo"
                src={undefined}
              />,
              <styled.h1>
                Welcome to React
              </styled.h1>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "logo",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Welcome to React",
              },
              "ref": null,
              "rendered": "Welcome to React",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "To get started, edit ",
              <code>
                src/App.tsx
              </code>,
              " and save to reload.",
            ],
          },
          "ref": null,
          "rendered": Array [
            "To get started, edit ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "src/App.tsx",
              },
              "ref": null,
              "rendered": "src/App.tsx",
              "type": "code",
            },
            " and save to reload.",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
